api/team.go

```diff
@@ -708,6 +709,19 @@ func findTeamByName(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getTeamByName(c *Context, w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	teamname := params["team_name"]
+
+	if result := <-Srv.Store.Team().GetByName(teamname); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else {
```

Comment:
Can you also add a case for this in the unit test? Essentially we want to make sure that if the team isn't open, then a user not on the team can't get it by name unless they are the system administrator